src/doc/rustdoc/src/write-documentation/re-exports.md

```diff
@@ -0,0 +1,128 @@
+```
+
+And now users will be able to do:
+
+```rust,ignore (inline)
+use lib::{Foo, AnotherFoo};
+```
+
+And since both `sub_module1` and `sub_module2` are private, users won't be able to import them.
+
+Now what's interesting is that the generated documentation for this crate will show both `Foo` and
+`AnotherFoo` directly at the crate root, meaning they have been inlined. There are a few rules to
+know whether or not a re-exported item will be inlined.
+
+## Inlining rules
+
+If a public item comes from a private module, it will be inlined:
+
+```rust,ignore (inline)
+mod private_module {
+    pub struct Public;
+}
+
+pub mod public_mod {
+    // `Public` will inlined here since `private_module` is private.
+    pub use super::private_module::Public;
+}
+
+// `Public` will not be inlined here since `public_mod` is public.
+pub use self::public_mod::Public;
+```
+
+Likewise, if an item has `#[doc(hidden)]` or inherits it (from any of its parents), it
+will be inlined:
+
+```rust,ignore (inline)
+#[doc(hidden)]
+pub mod public_mod {
+    pub struct Public;
+}
+
+#[doc(hidden)]
+pub struct Hidden;
+
+// `Public` be inlined since its parent (`public_mod`) has `#[doc(hidden)]`.
+pub use self::public_mod::Public;
+// `Hidden` be inlined since it has `#[doc(hidden)]`.
+pub use self::Hidden;
+```
+
+The inlining rules are a bit different for glob re-exports (`pub use x::*`) for `#[doc(hidden)]`
+types. If we take the previous example and then re-export like this:
+
+```rust,ignore (inline)
+pub use self::*; // It will not inline the `Hidden` struct.
+pub use self::public_mod::*; // It will inline the `Public` struct.
+```
+
+It only impacts elements that have the `#[doc(hidden)]` attributes. If it only inherits it, then it
+is inlined.
+
+## Inlining with `#[doc(inline)]`
+
+You can use the `#[doc(inline)]` attribute if you want to force an item to be inlined:
+
+```rust,ignore (inline)
+pub mod public_mod {
+    pub struct Public;
+}
+
+#[doc(inline)]
+pub use self::public_mod::Public;
+```
+
+With this code, even though `public_mod::Public` is public and present in the documentation, the
+`Public` type will be present both at the crate root and in the `public_mod` module.
+
+## Attributes
+
+When an item is inlined, its doc comments and most of its attributes will be inlined along with it:
+
+| Attribute | Inlined? | Notes
+|--|--|--
+| `#[doc=""]` | Yes | Intra-doc links are resolved relative to where the doc comment is defined (`///` is syntax sugar for doc string attributes).
+| `#[doc(cfg(..))]` | Yes |
+| `#[deprecated]` | Yes | Intra-doc links are resolved relative to where the description is defined.
+| `#[doc(alias="")]` | No |
+| `#[doc(inline)]` | No |
+| `#[doc(no_inline)]` | No |
+| `#[doc(hidden)]` | Glob imports | For name-based imports (such as `use module::Item as ModuleItem`), hiding an item acts the same as making it private. Glob-based imports (such as `use module::*`), hidden items are not inlined.
```

Comment:
this should also be mentioned on the docs for doc(hidden)

Reply:
:+1: